components/layout_2020/display_list/mod.rs

```diff
@@ -236,6 +242,139 @@ impl DisplayListBuilder<'_> {
             self.display_list.compositor_info.epoch.as_u16(),
         ))
     }
+
+    /// Draw highlights around the node that is currently hovered in the devtools
+    fn paint_dom_inspector_highlight(
+        &mut self,
+        highlighted_dom_node: OpaqueNode,
+        fragment_tree: &FragmentTree,
+    ) {
+        const CONTENT_BOX_HIGHLIGHT_COLOR: webrender_api::ColorF = webrender_api::ColorF {
+            r: 0.23,
+            g: 0.7,
+            b: 0.87,
+            a: 0.5,
+        };
+
+        const PADDING_BOX_HIGHLIGHT_COLOR: webrender_api::ColorF = webrender_api::ColorF {
+            r: 0.49,
+            g: 0.3,
+            b: 0.7,
+            a: 0.5,
+        };
+
+        const BORDER_BOX_HIGHLIGHT_COLOR: webrender_api::ColorF = webrender_api::ColorF {
+            r: 0.2,
+            g: 0.2,
+            b: 0.2,
+            a: 0.5,
+        };
+
+        const MARGIN_BOX_HIGHLIGHT_COLOR: webrender_api::ColorF = webrender_api::ColorF {
+            r: 1.,
+            g: 0.93,
+            b: 0.,
+            a: 0.5,
+        };
+
+        let mut content_box = euclid::Rect::default();
+        let mut maybe_box_fragment = None;
+        let tag_to_find = Tag::new(highlighted_dom_node);
+        fragment_tree.find(|fragment, _, containing_block| {
+            if fragment.tag() != Some(tag_to_find) {
+                return None::<()>;
+            }
+
+            let fragment_relative_rect = match fragment {
+                Fragment::Box(fragment) | Fragment::Float(fragment) => {
+                    maybe_box_fragment = Some((fragment.clone(), *containing_block));
+
+                    fragment.borrow().content_rect
+                },
+                Fragment::Positioning(fragment) => fragment.borrow().rect,
+                Fragment::Text(fragment) => fragment.borrow().rect,
+                Fragment::Image(image_fragment) => image_fragment.borrow().rect,
+                Fragment::AbsoluteOrFixedPositioned(fragment) => {
+                    fragment.borrow().static_position_rect
+                },
```

Comment:
I don't think you want to highlight this as it is the breadcrumb left over from hoisted fragments (which are represented as `Fragment::Box`). See the rustdoc for `Fragment::AbsoluteOrFixedPositioned`.